src/components/Graph/ControlPanel.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { CSSProperties, useState } from 'react';
+import { Node, Edge } from './types';
+import Button from '@mui/material/Button';
+import Legend from './Legend';
+import ScaleLegend from './ScaleLegend';
+import KeyboardDoubleArrowRightIcon from '@mui/icons-material/KeyboardDoubleArrowRight';
+import KeyboardDoubleArrowLeftIcon from '@mui/icons-material/KeyboardDoubleArrowLeft';
+import Stack from '@mui/material/Stack';
+import { Typography } from '@mui/material';
+
+interface ControlPanelProps {
+  toggles: { [key: string]: boolean };
+  onToggle: (category: string) => void;
+  simpleCategories: string[];
+  edgeType: boolean;
+  colorFunc?: (node: Node | Edge) => string;
+  elements: Node[];
+  edges: Edge[];
+  scales: number[];
+  scaleWidth: (n: number) => number;
+  downloadScreenshot: () => void;
+  randomize: () => void;
+  organize: () => void;
+  toggleLabels: () => void;
+  legendToggle?: (node: Node | Edge) => string;
+}
+
+const ControlPanel: React.FC<ControlPanelProps> = ({
+  toggles,
+  onToggle,
+  simpleCategories,
+  edgeType,
+  colorFunc,
+  elements,
+  edges,
+  scales,
+  scaleWidth,
+  downloadScreenshot,
+  randomize,
+  organize,
+  toggleLabels,
+  legendToggle,
+}) => {
+  const [collapsed, setCollapsed] = useState(false);
+
+  const panelStyle: CSSProperties = {
+    position: 'absolute',
+    top: '0',
+    right: '0',
+    width: collapsed ? '40px' : '250px',
+    backgroundColor: 'white',
+    transition: 'width 0.3s',
+    zIndex: 1000,
+    border: '1px solid grey',
+  };
+
+  const buttonStyle: CSSProperties = {
+    // left: collapsed ? '-12px' : '-10px',
+    width: collapsed ? '0px' : '10px',
+
+    height: '40px',
+    display: 'inline-block',
+    alignItems: 'left',
+    justifyContent: 'left',
+    cursor: 'pointer',
+    padding: '0px',
+  };
+  return (
+    <div style={panelStyle}>
```

Comment:
This div needs to be made scrollable vertically. Cuts off when not given the room. I'm not sure the exact CSS properties you need to change here to accomplish this, but check out this: https://github.com/weng-lab/SCREEN2.0/blob/main/screen2.0/src/app/search/page.tsx#L461

This is the parent Drawer Element of the Filters panel on screen which is a Scrollable and collapsible element. The Paper component being selected in sx is what's scrolling I think. I think it's probably something with position: 'absolute' or it's height not being set properly, but not positive